packages/gatsby-theme-docs/gatsby-node.js

```diff
@@ -147,31 +161,58 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
   if (navigationYamlResult.errors) {
     reporter.panicOnBuild('🚨  ERROR: Loading "allNavigationYaml" query');
   }
-  const pages = allMdxPagesResult.data.allMdx.edges;
+  const pages = allMdxPagesResult.data.contents.nodes;
   const navigationPages = navigationYamlResult.data.allNavigationYaml.nodes.reduce(
     (pageLinks, node) => [...pageLinks, ...(node.pages || [])],
     []
   );
-  pages.forEach(({ node }) => {
-    const matchingNavigationPage = navigationPages.find(
-      (page) =>
-        trimTrailingSlash(page.path) === trimTrailingSlash(node.fields.slug)
-    );
-    actions.createPage({
-      // This is the slug you created before
-      // (or `node.frontmatter.slug`)
-      path: node.fields.slug,
-      // This component will wrap our MDX content
-      component: require.resolve('./src/templates/page-content.js'),
-      // You can use the values in this context in
-      // our page layout component
-      context: {
-        ...node.fields,
-        shortTitle: matchingNavigationPage
-          ? matchingNavigationPage.title
-          : undefined,
-      },
-    });
+  pages.forEach(({ childMdx }) => {
+    if (childMdx) {
+      const matchingNavigationPage = navigationPages.find(
+        (page) =>
+          trimTrailingSlash(page.path) ===
+          trimTrailingSlash(childMdx.fields.slug)
+      );
+      actions.createPage({
+        // This is the slug you created before
+        // (or `node.frontmatter.slug`)
+        path: childMdx.fields.slug,
+        // This component will wrap our MDX content
+        component: require.resolve('./src/templates/page-content.js'),
+        // You can use the values in this context in
+        // our page layout component
+        context: {
+          ...childMdx.fields,
+          shortTitle: matchingNavigationPage
+            ? matchingNavigationPage.title
+            : undefined,
+        },
+      });
+    }
+  });
+
+  // create release notes overview page
+  actions.createPage({
+    path: '/release-notes',
+    component: require.resolve('./src/templates/release-notes.js'),
+  });
```

Comment:
This is not necessary, we can use an `index.mdx` in the `release-notes` folder.
Then in the `forEach` below you can use the "list overview" template for the `index.mdx` page, and the "detail page" template for all other pages.

Reply:
At least, from what I understood, in the `index.mdx` file you should be able to still write some intro content and then get the list of the release notes **below** that.

So the template for the release notes should query both the single page and all other pages.
The single page content should be rendered first, then the list of release notes.